rpcs3/evdev_joystick_handler.cpp

```diff
@@ -413,10 +413,51 @@ void evdev_joystick_handler::ThreadProc()
             break;
         }
         case EV_ABS:
-            LOG_NOTICE(GENERAL, "Joystick #%d EV_ABS: %d %d", i, evt.code, evt.value);
+        {
 
-            if (evt.code >= ABS_HAT0X && evt.code <= ABS_HAT3Y) {
-                int hat = evt.code - ABS_HAT0X;
+            LOG_NOTICE(GENERAL, "Joystick #%d EV_ABS: %d %d", i, evt.code, evt.value);
+            //duplicate the evt.code and evt.value, so that they're still available
+            //for the analog input
+            int code2=evt.code;
+            int value2=evt.value;
+            if (g_evdev_joystick_config.left_analog_to_dpad)
+            {
+                //some gamepad have a -32767 to +32767 range; others have a 0 to 255
+                float center2=(axis_ranges[evt.code].second+axis_ranges[evt.code].first)/2;
+                //arbitrary decision: when the value is higher than a 1/3 of the range, we simulate a dpad press
+                float threshold2=(axis_ranges[evt.code].second-center2)/3;
+                if (code2 == 0)
```

Comment:
not sure what's happening with this `if` `else if`, it looks to be the exact same code?
Why not just `if (code2 == 0 || code2 == 1)`

Reply:
This part is different:
``` 
if (g_evdev_joystick_config.lxstick<g_evdev_joystick_config.lystick)
                        code2=ABS_HAT0X;
                    else
                        code2=ABS_HAT0Y;
```
But, as you didn't see it, I rewrote it so that it is more explicit.
Thanks for your comments.